### PR TITLE
feat: use stats and include webpack runtime

### PR DESCRIPTION
**Pull request by mcasimir**

### CODE REVIEW

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts [hunk 1]
```diff
@@ -69,9 +71,44 @@ export class WebpackDependenciesPlugin implements WebpackPluginInstance {
         this.resolvedModules.add(packageJsonPath);
       }
     }
-  };
+  }
 
   apply(compiler: Compiler): void {
+    const webpackModulePath = this.getWebpackModulePath(new Error());
+    console.log(webpackModulePath);
```
Comment (gribnoysup) [inline packages/sbom-tools/src/webpack-dependencies-plugin.ts:78]:
```suggestion
```

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts [hunk 2]
```diff
@@ -69,9 +71,44 @@ export class WebpackDependenciesPlugin implements WebpackPluginInstance {
...
+      console.log(modules?.filter((m) => m.moduleType === 'runtime'));
+
+      modules?.forEach(({ type, nameForCondition }) => {
+        if (
+          type === 'module' &&
+          nameForCondition &&
+          this.isThirdPartyModule(nameForCondition)
+        ) {
+          console.log(nameForCondition);
+
```
Comment (gribnoysup) [inline packages/sbom-tools/src/webpack-dependencies-plugin.ts:94]:
```suggestion
```

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts [hunk 3]
```diff
@@ -69,9 +71,44 @@ export class WebpackDependenciesPlugin implements WebpackPluginInstance {
...
+    const webpackModulePath = this.getWebpackModulePath(new Error());
+    console.log(webpackModulePath);
+
+    this.addIncludedPackages(compiler);
+
+    compiler.hooks.done.tapAsync(PLUGIN_NAME, (stats, done) => {
+      const { modules } = stats.toJson();
+
+      console.log(modules?.filter((m) => m.moduleType === 'runtime'));
+
```
Comment (gribnoysup) [inline packages/sbom-tools/src/webpack-dependencies-plugin.ts:86]:
```suggestion
```

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts [hunk 4]
```diff
@@ -69,9 +71,44 @@ export class WebpackDependenciesPlugin implements WebpackPluginInstance {
...
+        ) &&
+        webpackModulePath
+      ) {
+        this.resolvedModules.add(webpackModulePath);
+      }
+
+      done();
+    });
+
     compiler.hooks.shutdown.tapPromise(PLUGIN_NAME, async () => {
```
Comment (gribnoysup) [inline packages/sbom-tools/src/webpack-dependencies-plugin.ts:112]: Doesn't make that much of a difference, but this probably can be just done in the same stats hook? I wonder if you kept it in this one for some particular reason 🤔

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts (hunk 1 above)
Comment (mcasimir) [reply packages/sbom-tools/src/webpack-dependencies-plugin.ts:78]: WIP sorry, i'm fixing the CI

File: packages/sbom-tools/src/webpack-dependencies-plugin.ts (hunk 4 above)
Comment (gribnoysup) [reply packages/sbom-tools/src/webpack-dependencies-plugin.ts:107]: Multi-config, right, makes sense!

**Pull request merged**